cli/src/main/java/org/jboss/as/cli/logger/CliLogger.java

```diff
@@ -0,0 +1,235 @@
+package org.jboss.as.cli.logger;
+
+import org.aesh.readline.terminal.formatting.TerminalString;
+import org.jboss.dmr.ModelNode;
+import org.jboss.logging.Logger;
+import org.jboss.logging.annotations.Cause;
+import org.jboss.logging.annotations.LogMessage;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageLogger;
+
+import java.util.Collection;
+import java.util.List;
+
+@MessageLogger(projectCode = "WFLYCLI")
+public interface CliLogger {
```

Comment:
This should extend the `BasicLogger` so that you can use debug and trace logging from the `CliLogger`.

Reply:
Two other general comments. I don't like the `BASE + x`. I'd also prefer using the default `PRINTF` format over the `MESSAGE_FORMAT`. It's much more robust.

Reply:
Also trace and debug logging don't need to be in here. Those are not required to be internationalized.

Reply:
@jamezp thank you for the review. Would you prefer me to move DEBUG/TRACE messages out of CliLogger?

Reply:
It doesn't matter too much to me, but the rest of WildFly Core and WildFly loggers do not include DEBUG and TRACE messages in the interface. If you extend the `BasicLogger` interface you can just use `CliLogger.ROOT_LOGGER.debugf("Some debug message with %s", parameter)`.